### PR TITLE
Add Dockerfile and publish the official Docker image

### PR DESCRIPTION
**Pull request by abhi18av**
Hi @jfacorro ,

I've been following the project for a while now and I thought that it might help to have a docker image ready for the quickly trying out the project. I noticed that you previously did make some initiative on `docker hub` here https://hub.docker.com/r/jfacorro/clojerl but it doesn't seem to be usable.

The `Dockerfile` I've added at the moment is quite basic and so, _I've only created a draft PR asI'm looking forward to adapting the PR as to how you see fit_.

I'd really like to see `official clojerl` docker image to be there on `docker hub`.

### CODE REVIEW

**Comment by abhi18av** [comment]
Here are certain improvements on my mind

[ ] Linking the `Docker hub` with the `CD` pipeline for this repo
[ ] Perhaps pin the base `erlang` image to a specific version
[ ] Updating the Readme

**Comment by jfacorro** [comment]
@abhi18av Thank you for the contribution! 

I agree with the points you list above. I have created a clojerl account in docker hub [here](https://hub.docker.com/repository/docker/clojerl/clojerl), so that once this get merge there's already a place where it can be published.

I built the `Dockerfile` included in this PR and I have only one reuqest, which is that instead of using `bin/clojerl` as the entrypoint we use `bin/clje`. This will require that `rlwrap` is installed but IMO the user experience will be much better. Here's a stab at what using `bin/clje` would involve:
```
FROM erlang:latest

RUN git clone https://github.com/clojerl/clojerl && \
        cd clojerl                               && \
        make                                     && \
        apt-get update                           && \
        apt-get install rlwrap                   && \
        apt-get clean                            && \
        apt-get autoclean

CMD ["/clojerl/bin/clje"]
```

**Comment by jfacorro** [comment]
Sadly, there is an issue when using `rlwrap` in a docker container as an entrypoint, see [here](https://github.com/sflyr/docker-sqlplus/pull/2). So I guess we either add the sleep (as below) or keep the original approach of using `/bin/clojerl`.

```
FROM erlang:latest

RUN git clone https://github.com/clojerl/clojerl && \
        cd clojerl                               && \
        make                                     && \
        apt-get update                           && \
        apt-get install rlwrap                   && \
        apt-get clean                            && \
        apt-get autoclean

CMD sleep 0.1; /clojerl/bin/clje
```

**Comment by abhi18av** [comment]
Thanks for the quick revert @jfacorro , I've updated the dockerfile as you mentioned and I'll be updating the `Readme` in a bit.

File: README.md [hunk 1]
```diff
@@ -32,6 +32,25 @@ what [features][features] does it include and
...
 [Try Clojerl][try-clojerl].
 
+## Docker REPL
+
+To quickly try out `clojerl` via docker you could make use of docker image like so 
+
+```
+docker pull clojerl/clojerl
+docker run -it clojerl/clojerl
+
```
Comment (jfacorro) [inline README.md]: Nit: This extra new line seems to show up when the markdown is rendered.

File: scripts/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
+FROM erlang:latest
```
Comment (jfacorro) [inline scripts/Dockerfile]: I think it would be good to pin this to a specific version as you had mentioned in the comments. `22.2.3` should work.

Comment (abhi18av) [reply scripts/Dockerfile]: Yup, done 👍

**Comment by jfacorro** [comment]
Merged 🎉 ! Thank you again for the contribution 😄

**Comment by abhi18av** [comment]
You're most welcome @jfacorro 😊

**Comment by jfacorro** [comment]
The image is now available [here](https://hub.docker.com/repository/docker/clojerl/clojerl/tags?page=1).

**Pull request merged**